tests/core/test_device_save_loader.py

```diff
@@ -0,0 +1,197 @@
+async def test_save_device_with_phase(device_with_phases, tmp_path):
+    RE = RunEngine()
+    await device_with_phases.child1.sig1.set("string")
+    table_pv = {"VAL1": np.array([1, 1, 1, 1, 1]), "VAL2": np.array([1, 1, 1, 1, 1])}
+    await device_with_phases.child2.sig1.set(table_pv)
+
+    # Create save plan from utility functions
+    def save_my_device():
+        signalRWs = walk_rw_signals(device_with_phases)
+        values = yield from get_signal_values(signalRWs)
+        phases = device_with_phases.sort_signal_by_phase(device_with_phases, values)
+        save_to_yaml(phases, path.join(tmp_path, "test_file.yaml"))
+
+    RE(save_my_device())
+
+    with open(path.join(tmp_path, "test_file.yaml"), "r") as file:
+        yaml_content = yaml.load(file, yaml.Loader)
+        assert yaml_content[0] == {"child1.sig1": "string"}
+        assert np.array_equal(
+            yaml_content[1]["child2.sig1"]["VAL1"], np.array([1, 1, 1, 1, 1])
+        )
+        assert np.array_equal(
+            yaml_content[1]["child2.sig1"]["VAL2"], np.array([1, 1, 1, 1, 1])
+        )
+
+
+async def test_yaml_formatting_no_phase(device_with_phases, tmp_path):
+    RE = RunEngine()
+    await device_with_phases.child1.sig1.set("test_string")
+    table_pv = {"VAL1": np.array([1, 2, 3, 4, 5]), "VAL2": np.array([6, 7, 8, 9, 10])}
+    await device_with_phases.child2.sig1.set(table_pv)
+
+    # Create save plan from utility functions
+    def save_my_device():
+        signalRWs = walk_rw_signals(device_with_phases)
+        values = yield from get_signal_values(signalRWs)
+        phases = device_with_phases.sort_signal_by_phase(device_with_phases, values)
+        save_to_yaml(phases, path.join(tmp_path, "test_file.yaml"))
+
+    RE(save_my_device())
+
+    with open(path.join(tmp_path, "test_file.yaml"), "r") as file:
+        expected = """\
+- {child1.sig1: test_string}
+- child2.sig1:
+    VAL1: [1, 2, 3, 4, 5]
+    VAL2: [6, 7, 8, 9, 10]
+"""
+        assert file.read() == expected
+
+
+async def test_saved_types_with_phase(device_with_phases, tmp_path):
+    RE = RunEngine()
+    await device_with_phases.child1.sig1.set("string")
+    table_pv = {"VAL1": np.array([1, 1, 1, 1, 1]), "VAL2": np.array([1, 1, 1, 1, 1])}
+    await device_with_phases.child2.sig1.set(table_pv)
+
+    # Create save plan from utility functions
+    def save_my_device():
+        signalRWs = walk_rw_signals(device_with_phases)
+        values = yield from get_signal_values(signalRWs)
+        phases = device_with_phases.sort_signal_by_phase(device_with_phases, values)
+        save_to_yaml(phases, path.join(tmp_path, "test_file.yaml"))
+
+    RE(save_my_device())
+
+    with open(path.join(tmp_path, "test_file.yaml"), "r") as file:
+        yaml_content = yaml.load(file, yaml.Loader)
+        assert type(yaml_content[0]["child1.sig1"]) is str
+        assert type(yaml_content[1]["child2.sig1"]["VAL1"]) is list
+        assert type(yaml_content[1]["child2.sig1"]["VAL2"]) is list
```

Comment:
not sure about the language here r.e. phases. It seems like these three tests share a lot in common, and can probably be combined into one test and still cover the same lines.

to me, these tests only differ from the previous few because they test if save_to_yaml can handle a List[Dict[str, Any]] as well as a Dict[str, Any]. Which is fine, but why the word 'phase'? Would changing 'phase' to 'list' be better in this instance?

Reply:
Yep, I can reduce these tests down. I believe the idea with "phases" is that when loading values from a yaml to a Device (which is yet to be implemented) we want to load the groups of signals in a specified order, so all the phase 1 signals first, then phase 2 signals etc. There's possibly a clearer way to express this.